### PR TITLE
Use digest.md5_hex to compute md5 digests instead of openssl

### PR DESCRIPTION
**Pull request by knazarov**
luarocks has support for calculating md5 with 'md5' rock if it's
present, but we don't have it in tarantool, and instead have the
'digest' module. That's why luarocks falls back to 'openssl' binary to
calculate md5 digests.

This patch will allow luarocks to use our internal digests module.

### CODE REVIEW

**Comment by Totktonada** [comment]
Should be merged into `tarantool-1.7`, not `master`.

**Comment by knazarov** [comment]
@Totktonada yeah, nice catch. I've rebased the patch to correct branch.

**Comment by knazarov** [comment]
Also, I've checked it on environment without openssl, and it seems to work fine.

File: src/luarocks/fs/lua.lua [hunk 1]
```diff
@@ -11,16 +11,16 @@ local dir = require("luarocks.dir")
...
+local http, ftp, lrzip, luazip, lfs, digest, posix
 
 if cfg.fs_use_modules then
    socket_ok, http = pcall(require, "socket.http")
    _, ftp = pcall(require, "socket.ftp")
    zip_ok, lrzip = pcall(require, "luarocks.tools.zip")
    unzip_ok, luazip = pcall(require, "zip"); _G.zip = nil
    lfs_ok, lfs = pcall(require, "lfs")
-   md5_ok, md5 = pcall(require, "md5")
+   digest_ok, digest = pcall(require, "digest")
```
Comment (Gerold103) [inline src/luarocks/fs/lua.lua:24]: How is it possible that digest_ok is false?

File: src/luarocks/fs/lua.lua [hunk 2]
```diff
@@ -11,16 +11,16 @@ local dir = require("luarocks.dir")
...
    posix_ok, posix = pcall(require, "posix")
```
Comment (Gerold103) [inline src/luarocks/fs/lua.lua:25]: If you care about compatibility with future Tarantool versions, where digest could be removed, then why no to check for md5 if digest is not ok?
```Lua
-- ...
if not digest_ok then
    md5_ok, md5 = pcall(require, "md5")
end
-- ...
```

**Review by Gerold103 — CHANGES_REQUESTED** [review]

**Comment by rosik** [comment]
As I've already mentioned IRL, I don't like this solution. 
Despite all other changes in tarantool/luarocks fork, it is still compatible with native lua.

Initially my proposal was to hack tarantoolctl in tarantool repo instead of luarocks: https://github.com/tarantool/tarantool/commit/23f25da77c1e5f30373290110f6af69246cf9cd4

Now I suggest an alternative more clean way: https://github.com/tarantool/tarantool/commit/ec5bd45a27c030a8b43becd12ce20a1e8c95c3bd

**Comment by Totktonada** [comment]
I think we can just check both `digest` and `md5` modules in luarocks and do not perform monkey patching or some other hacking in tarantoolctl.

I'm against of the hack, because a reader of luarocks code would not even have an idea what is going on if, say, `digest.md5_hex` returns a wrong value. Having it explicit allow to understand what the code do.

File: src/luarocks/fs/lua.lua (hunk 2 above)
Comment (knazarov) [reply src/luarocks/fs/lua.lua:25]: I've updated the patch to use both ways. Hope it ends the discussion

File: src/luarocks/fs/lua.lua (hunk 1 above)
Comment (knazarov) [reply src/luarocks/fs/lua.lua:24]: @Totktonada and @rosik insist that we don't break compatibility with regular lua. I don't see any major reason to do so, but anyway...

**Review by knazarov — COMMENTED** [review]
Fixed the code to make implementation less "intrusive"

**Review by Gerold103 — APPROVED** [review]

**Review by Totktonada — APPROVED** [review]

**Pull request merged**